xbmc/peripherals/devices/Peripheral.cpp

```diff
@@ -149,10 +150,21 @@ bool CPeripheral::Initialise(void)
     return bReturn;
 
   g_peripherals.GetSettingsFromMapping(*this);
-  m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s_%s.xml",
-                                          PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
-                                          m_strVendorId.c_str(),
-                                          m_strProductId.c_str());
+  if (m_iVendorId == 0x0000 && m_iProductId == 0x0000)
+  {
+    m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s.xml",
+                                            PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
+                                            CUtil::MakeLegalFileName(m_strDeviceName, LEGAL_WIN32_COMPAT).c_str());
+  }
+  else
+  {
+    m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s_%s_%s.xml",
+                                            PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
+                                            m_strVendorId.c_str(),
+                                            m_strProductId.c_str(),
+                                            CUtil::MakeLegalFileName(m_strDeviceName, LEGAL_WIN32_COMPAT).c_str());
```

Comment:
This changes the name of the RPi CEC settings file from `rpi_2708_1001.xml` to `rpi_2708_1001_CEC Adapter.xml` which is going to be a small support headache when users upgrading to Kodi 17 complain their CEC isn't working (because they're now back to default settings).